src/jrd/tpc.cpp

```diff
@@ -375,14 +405,30 @@ TipCache::StatusBlockData::~StatusBlockData()
 void TipCache::StatusBlockData::clear(thread_db* tdbb)
 {
 	// memory could be already released at tpc_block_blocking_ast
+	PathName fName;
 	if (memory)
 	{
-		memory->removeMapFile();
+		// wait for all initializing processes (PR)
+		if (!LCK_convert(tdbb, &existenceLock, LCK_SW, LCK_WAIT))
+			ERR_bugcheck_msg("Unable to convert TPC lock (SW)");
+
+		fName = memory->getMapFileName();
 		delete memory;
 		memory = NULL;
 	}
 
 	LCK_release(tdbb, &existenceLock);
+
+	if (fName.hasData())
+	{
+		if (LCK_lock(tdbb, &existenceLock, LCK_EX, LCK_NO_WAIT))
+		{
+			SharedMemoryBase::unlinkFile(fName.c_str());
+			LCK_release(tdbb, &existenceLock);
+		}
+		else
+			tdbb->tdbb_status_vector->init();
+	}
```

Comment:
Here same optimization as in `TipCache::finalizeTpc()` could be applied - don't call `LCK_release` before `LCK_lock`.
I'm not insist on it, though.